### PR TITLE
[TEST] re-enable qa tests for mapped runtime fields

### PR DESCRIPTION
**Pull request by javanna**
These tests were left disabled since the introduction of the runtime section, as dynamic templates could not yet create a runtime field. Now that dynamic templates can do that, we re-enable them, though we lose a bit of coverage as we have no ability to mimic the multi-field that's created when string fields are dynamically mapped.

### CODE REVIEW

File: x-pack/plugin/runtime-fields/qa/src/main/java/org/elasticsearch/xpack/runtimefields/test/CoreTestTranslater.java [hunk 1]
```diff
@@ -79,18 +80,18 @@
         NumberType.DOUBLE.typeName(),
         KeywordFieldMapper.CONTENT_TYPE,
         IpFieldMapper.CONTENT_TYPE,
-        NumberType.LONG.typeName()
+        NumberType.LONG.typeName(),
+        GeoPointFieldMapper.CONTENT_TYPE
```
Comment (javanna) [inline x-pack/plugin/runtime-fields/qa/src/main/java/org/elasticsearch/xpack/runtimefields/test/CoreTestTranslater.java:84]: this is triggering new failures, so I guess I am better off making this change separately.

**Comment by javanna** [comment]
We do have two legit failures, @nik9000 do those look familiar to you?

```
org.elasticsearch.xpack.runtimefields.test.mapped.CoreWithMappedRuntimeFieldsIT.test {yaml=search.aggregation/40_range/Date range}
org.elasticsearch.xpack.runtimefields.test.mapped.CoreWithMappedRuntimeFieldsIT.test {yaml=search.aggregation/220_filters_bucket/cache}
```

**Comment by nik9000** [comment]
The second one fails because we don't support `split_queries_on_whitespace` in runtime fields. I think we probably should actually.

**Comment by nik9000** [comment]
The first one I don't know what is up.

File: x-pack/plugin/runtime-fields/qa/src/main/java/org/elasticsearch/xpack/runtimefields/test/CoreTestTranslater.java [hunk 2]
```diff
@@ -109,24 +110,19 @@ public void execute(ClientYamlTestExecutionContext executionContext) throws IOEx
                 Map<String, String> params = Map.of("name", "hack_dynamic_mappings", "create", "true");
                 List<Map<String, Object>> dynamicTemplates = new ArrayList<>();
                 for (String type : RUNTIME_TYPES) {
-                    if (type.equals("ip")) {
-                        // There isn't a dynamic template to pick up ips. They'll just look like strings.
+                    if (type.equals(IpFieldMapper.CONTENT_TYPE)
+                        || type.equals(GeoPointFieldMapper.CONTENT_TYPE)
+                        || type.equals(KeywordFieldMapper.CONTENT_TYPE)) {
+                        // There isn't a dynamic template to pick up ips and geopoints. They'll just look like strings.
```
Comment (nik9000) [inline x-pack/plugin/runtime-fields/qa/src/main/java/org/elasticsearch/xpack/runtimefields/test/CoreTestTranslater.java:116]: It's probably worth a special comment for `keyword` here.

Comment (javanna) [reply x-pack/plugin/runtime-fields/qa/src/main/java/org/elasticsearch/xpack/runtimefields/test/CoreTestTranslater.java]: I had already written a comment for keyword one line below, anyways I updated those comments.

File: x-pack/plugin/runtime-fields/qa/build.gradle [hunk 3]
```diff
@@ -57,15 +55,17 @@ subprojects {
...
           'field_caps/30_filter/Field caps with index filter', // We don't support filtering field caps on runtime fields. What should we do?
-          'search.aggregation/10_histogram/*', // runtime doesn't support sub-fields. Maybe it should?
+          'search.aggregation/220_filters_bucket/cache', // runtime keyword does not support split_queries_on_whitespace
           'search/140_pre_filter_search_shards/pre_filter_shard_size with shards that have no hit',
           /////// TO FIX ///////
 
           /////// NOT SUPPORTED ///////
+          'search.highlight/40_keyword_ignore/Plain Highligher should skip highlighting ignored keyword values', // The plain highlighter is incompatible with the prefix queries that we make for runtime fields, use unified highlighter instead.
           'search.aggregation/280_rare_terms/*', // Requires an index and we won't have it
+          'search.aggregation/10_histogram/*', // Runtime fields don't support sub-fields
```
Comment (javanna) [inline x-pack/plugin/runtime-fields/qa/build.gradle:68]: I moved a couple of tests down here because I am sure we won't fix them.

**Comment by javanna** [comment]
@nik9000 I curated the list of exclusions, tests are now green.

File: x-pack/plugin/runtime-fields/qa/build.gradle (hunk 3 above)
Comment (nik9000) [reply x-pack/plugin/runtime-fields/qa/build.gradle:68]: :+1:

**Review by nik9000 — APPROVED** [review]
LGTM

Thanks for all that!

**Pull request merged**